src/commands/force/user/create.ts

```diff
@@ -54,36 +54,48 @@ export class UserCreateCommand extends SfdxCommand {
   public org: Org;
 
   private user: User;
-  private successes: SuccessMsg[];
-  private failures: FailureMsg[];
+  private successes: SuccessMsg[] = [];
+  private failures: FailureMsg[] = [];
 
   public async run(): Promise<UserFields> {
     this.logger = await Logger.child(this.constructor.name);
     const defaultUserFields: DefaultUserFields = await DefaultUserFields.create({
       templateUser: this.org.getUsername(),
     });
-    const user: User = await User.create({ org: this.org });
+    this.user = await User.create({ org: this.org });
 
     // merge defaults with provided values with cli > file > defaults
     const fields: UserFields & Dictionary<string> = await this.aggregateFields(defaultUserFields.getFields());
+    // because fields is type UserFields & Dictionary<string> we can access these
+    const permsets = fields['permsets'];
+    const generatepassword: string = fields['generatepassword'];
+
+    // extract the fields and then delete, createUser doesn't expect a permsets or generatepassword
+    delete fields['permsets'];
+    delete fields['generatepassword'];
+    delete fields['generatePassword'];
 
     try {
-      await user.createUser(fields);
+      await this.user.createUser(fields);
     } catch (e) {
       await this.catchCreateUser(e, fields);
     }
 
-    // because fields is type UserFields & Dictionary<string> we can access these
-    const permsets: string = fields['permsets'];
-    const generatepassword: string = fields['generatepassword'];
-
     // Assign permission sets to the created user
     if (permsets) {
       try {
-        await this.user.assignPermissionSets(fields.id, permsets.trim().split(','));
```

Comment:
This does not seem equivalent. If the permset is a string, it can still be a comma-separated array right? I think you still want that here.

```
const permsets: string | string[] = fields.permsets; // you might have to fix the type of fields or cast this to unknown then string | string[].
const permsetArray = isArray<string>(permsets) ? permsets : permsets.trim().split(',');
```